core/domain/exp_services.py

```diff
@@ -957,37 +959,45 @@ def update_exploration(
             exploration_id)
 
 
-def regenerate_exploration_summary(exploration_id, contributor_id_to_add):
-    """Regenerate a summary of the given exploration. If the summary does not
-    exist, this function generates a new one.
+def regenerate_exploration_summary_with_new_contributor(
+        exploration_id, contributor_id):
+    """Regenerate a summary of the given exploration and add a new contributor
+    to the contributors summary. If the summary does not exist, this function
+    generates a new one.
 
     Args:
         exploration_id: str. The id of the exploration.
-        contributor_id_to_add: str or None. The user_id of user who have
-            created the exploration will be added to the list of contributours
-            for the exploration if the argument is not None and it is not a
-            system id.
+        contributor_id: str. ID of the contributor to be added to
+            the exploration summary.
     """
     exploration = exp_fetchers.get_exploration_by_id(exploration_id)
-    exp_summary = compute_summary_of_exploration(
-        exploration, contributor_id_to_add)
+    exp_summary = _compute_summary_of_exploration(exploration)
+    exp_summary.add_new_contribution_for_user(contributor_id)
     save_exploration_summary(exp_summary)
 
 
-def compute_summary_of_exploration(exploration, contributor_id_to_add):
+def regenerate_exploration_and_contributors_summaries(exploration_id):
+    """Regenerate a summary of the given exploration and also regenerate
+    the contributors summary from the snapshots. If the summary does not exist,
+    this function generates a new one.
+
+    Args:
+        exploration_id: str. ID of the exploration.
+    """
+    exploration = exp_fetchers.get_exploration_by_id(exploration_id)
+    exp_summary = _compute_summary_of_exploration(exploration)
+    exp_summary.contributors_summary = (
+        compute_exploration_contributors_summary(exp_summary.id))
+    save_exploration_summary(exp_summary)
```

Comment:
Does this not need to happen in transaction? What happens if any of these objects change while it's being saved?